### ISSUE TITLE
Few Codebase improvements

### ISSUE THREAD

**Issue by gethari** [sendgrid/sendgrid-csharp#928]
#### Issue Summary

I have ran through the repo, and found some of my findings here. If my findings are valid i will go ahead to send a PR.

### Code style
The Qualifier `this` is redundant and there are 61 occurrences. Some people consider this as different. But ill leave this according to you as this is just a visual overhaul.

- All the models must fall under `SendGrid.Helpers.Mail.Models` instead of `SendGrid.Helpers.Mail`

### Unused methods and properties
In the `ISendGridClient` interface properties like UrlPath,Version,MediaType,AddAuthorization and the methods like `DeserializeResponseBodyAsync` and `DeserializeResponseBodyAsync`

### Test project

The test project is referencing `.NET CORE 1.0` . We should migrate it to the latest version of `.NET Core` atleast to 2.2.

And the `Integration.cs` is crossing 6K lines now, we should partition the file and split it so it would be easier to move forward

**Comment by thinkingserious** [comment]
Hello @gethari,

Thanks for contributing to our C# community! We certainly welcome PRs and in this case a sample PR would be a great start to get the conversation started.

With best regards,

Elmer

**Comment by gethari** [comment]
@thinkingserious any update on this ?

**Comment by jnyrup** [comment]
@gethari Are you aware of the consequences to your suggestions?

* Removing a member from an interface is a breaking change.
```c#
ISendGridClient myClient = ...
var urlPath = myClient.UrlPath; 
// No longer compile as the property is removed from the interface
```

* Changing the namespace a type is a breaking change
```c#
SendGrid.Helpers.Mail.Attachment attachment = new Attachment(); 
// No longer compiles as the namespace has changed
```

* Bumping the target framework will decrease the number of users that can consume the package. 
In most cases libraries should target the lowest possible framework. 
What problem do you seek to solve by bumping the requirement from .net core 1.0 to 2.1?

**Comment by thinkingserious** [comment]
Hello @gethari,

I'm closing this issue based on the referenced PR being closed as a duplicate of #981.

If you would like to contribute a separate PR with only the codebase improvements, we would greatly appreciate it.

With best regards,

Elmer

**Comment by thinkingserious** [comment]
@jnyrup,

Thanks for taking the time to review the changes and provide feedback!
